design/ceph/node-loss-rbd-cephfs.md

```diff
@@ -0,0 +1,150 @@
+
+For Cephfs recovery:
+
+Currently, Cephfs recovery is slower in case of node loss.
+
+## Proposed Solution
+
+> Note: This solution requires minimum kubernetes version 1.26.0
+
+The kubernetes feature [Non-Graceful Node Shutdown](https://kubernetes.io/blog/2022/12/16/kubernetes-1-26-non-graceful-node-shutdown-beta/) is available starting in Kubernetes 1.26 to help improve the volume recovery during node loss. When a node is lost, the admin is required to add the taint `out-of-service` manually to the node. After the node is tainted, Kubernetes will:
+
+- Remove the volume attachment from the lost node
+- Delete the old pod on the lost node
+- Create a new pod on the new node
+- Allow the volume to be attached to the new node
+
+Once this taint is applied manually, Rook will create a [NetworkFence CR](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/networkfence.md). The [csi-addons operator](https://github.com/csi-addons/kubernetes-csi-addons) will then blocklist the node to prevent any ceph rbd/cephfs client on the lost node from writing any more data.
+
+After the new pod is running on the new node and the old node which was lost comes back, Rook will delete the [NetworkFencr CR](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/networkfence.md).
+
+example of taint to be applied to lost node:
+
+```console
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoExecute
+# or
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoSchedule
+```
+
+> Note: This will be enabled by default in Rook if the NetworkFence CR is found, in the case for some reason user wants to disable this feature in Rook can edit `Rook-ceph-operator-config` configmap and update the `Rook_WATCH_FOR_NODE_FAILURE: "false"`.
+
+## How to get which IP to blocklist
+
+There are multiple networking options available for example, Host Networking, Pod networking, Multus etc. This make it difficult to know which NodeIP address to blocklist.
+For this we'll follow the following approach which will work for all networking options(but still will not work for external cluster which NodeIPs):
+
+1. Get the `volumesInUse` from the node which has the taint `out-of-service`.
+
+2. List all the pv and compare the pv `spec.volumeHandle` with the node `volumesInUse` field `volumeHandle`
+   1. Example:
+
+      Below is sample Node volumeInUse field
+
+    ```
+      volumesInUse:
+        - kubernetes.io/csi/Rook-ceph.rbd.csi.ceph.com^0001-0009-Rook-ceph-0000000000000002-24862838-240d-4215-9183-abfc0e9e4002
```

Comment:
```suggestion
kubernetes.io/csi/rook-ceph.rbd.csi.ceph.com^0001-0009-Rook-ceph-0000000000000002-24862838-240d-4215-9183-abfc0e9e4002`
```

Reply:
why  '`' in the end?

Reply:
thats not required, we can discard it. i missed adding ``` at start when adding the comment